File: apps/ios/ADHDFocus/Models/Task.swift
import Foundation

enum TaskStatus: String, Codable, CaseIterable {
    case inbox
    case today
    case scheduled
    case inProgress = "in_progress"
    case done
    case archived
}

enum TaskPriority: String, Codable, CaseIterable {
    case must
    case should
    case want
    case someday
}

enum EnergyLevel: String, Codable, CaseIterable {
    case low
    case medium
    case high
}

struct TaskItem: Identifiable, Codable {
    let id: String
    var title: String
    var description: String?
    var status: TaskStatus
    var priority: TaskPriority?
    var energyRequired: EnergyLevel?
    var estimatedMinutes: Int?
    var actualMinutes: Int?
    var pomodorosCompleted: Int?
    var dueDate: String?
    var scheduledDate: String?
    var projectId: String?
    var completedAt: String?
    var tags: [String]?
    var sortOrder: Int?
    var createdAt: String
    var updatedAt: String
}

struct CreateTaskInput: Codable {
    var title: String
    var description: String?
    var status: TaskStatus?
    var priority: TaskPriority?
    var energyRequired: EnergyLevel?
    var estimatedMinutes: Int?
    var dueDate: String?
    var scheduledDate: String?
    var projectId: String?
    var tags: [String]?
}

struct UpdateTaskInput: Codable {
    var title: String?
    var description: String?
    var status: TaskStatus?
    var priority: TaskPriority?
    var energyRequired: EnergyLevel?
    var estimatedMinutes: Int?
    var actualMinutes: Int?
    var dueDate: String?
    var scheduledDate: String?
    var projectId: String?
    var tags: [String]?
    var sortOrder: Int?
}

File: apps/mobile/types/task.ts
export type TaskStatus = 'inbox' | 'today' | 'scheduled' | 'in_progress' | 'done' | 'archived';
export type EnergyLevel = 'low' | 'medium' | 'high';
export type Priority = 'must' | 'should' | 'want' | 'someday';

export interface Task {
  id: string;
  userId: string;
  title: string;
  description: string | null;
  status: TaskStatus | null;
  energyRequired: EnergyLevel | null;
  priority: Priority | null;
  estimatedMinutes: number | null;
  actualMinutes: number | null;
  pomodorosCompleted: number | null;
  dueDate: string | null;
  scheduledDate: string | null;
  completedAt: string | null;
  snoozedUntil: string | null;
  projectId: string | null;
  parentTaskId: string | null;
  tags: string[] | null;
  streakContribution: boolean | null;
  sortOrder: number | null;
  createdAt: string | null;
  updatedAt: string | null;
}

export interface CreateTaskInput {
  title: string;
  description?: string;
  energyRequired?: EnergyLevel;
  priority?: Priority;
  estimatedMinutes?: number;
  dueDate?: string;
  scheduledDate?: string;
  projectId?: string;
  status?: 'inbox' | 'today' | 'scheduled';
}

export interface UpdateTaskInput {
  title?: string;
  description?: string | null;
  status?: TaskStatus;
  energyRequired?: EnergyLevel;
  priority?: Priority;
  estimatedMinutes?: number | null;
  dueDate?: string | null;
  scheduledDate?: string | null;
  projectId?: string | null;
  completedAt?: string | null;
  sortOrder?: number;
}
